src/main/java/com.outsystems.plugins.inappbrowser/osinappbrowserlib/helpers/OSIABCustomTabsSessionHelper.kt

```diff
@@ -60,6 +60,8 @@ class OSIABCustomTabsSessionHelper: OSIABCustomTabsSessionHelperInterface {
     ) : CustomTabsCallback() {
 
         private var isCustomTabsActivityOnTop = false
+        private var pendingTabHiddenEvent = false
```

Comment:
What was causing the race condition and what was the approach to fix it?

Reply:
There was a race condition with the `onBrowserFinished` event that I noticed when testing the plugin some more. 

Basically, when the invisible `CustomTabsActivity` receives an `onResume` event.. it will dispatch that to the session helper so `isCustomTabsActivityOnTop` gets set correctly (which we need for determining if the browser was hidden via the recent apps or because it was destroyed since Google only provides a `TAB_HIDDEN` event and not one for when the custom tabs is destroyed.). However, I noticed that the `TAB_HIDDEN` event was sometimes triggering before the `isCustomTabsActivityOnTop` flag is set.

To fix it, if we detect a `TAB_HIDDEN` event and the `isCustomTabsActivityOnTop` flag is not true.. we don't automatically assume that it's not a browser finished event. Instead, we set a `pendingTabHiddenEvent` flag to true and then we can detect the Browser finished properly if the `resume` event gets received after.